### ISSUE TITLE
"min" setting on collapsable group not working since 3.1.0

### ISSUE THREAD

**Issue by rockberte** [maxwroc/battery-state-card#637]
**Describe the bug**
I use the "min" setting on a group object to exclude states below 20% from the group.
It doesn't seem to be deterministic which states are excluded from the group.
In the case shown below there are three states with > 20% that should be included in the group.

**How to reproduce**

**Expected behavior**

**YAML configuration**
```yaml

type: custom:battery-state-card
title: Erdgeschoss
sort:
  by: state
color_gradient:
  - '#ff0000'
  - '#ffff00'
  - '#00ff00'
collapse:
  - name: '{count} Bewegungsmelder: {range}%'
    icon: mdi:motion-sensor
    min: 20
    entities:
      - sensor.bewegungsmelder_diele_eg_batterie
      - sensor.bewegungsmelder_flur_battery
  - name: '{count} Fenster: {range}%'
    icon: mdi:window-open-variant
    min: 20
    entities:
      - sensor.fenster_klein_buro_batterie
      - sensor.fenster_gr_buro_power
      - sensor.fenster_hauswirtschaftsraum_power
      - sensor.fenster_bad_eg_power
      - sensor.fenster_wc_power
      - sensor.fenster_kuche_power
      - sensor.terrassentur_wohnzimmer_power
      - sensor.haustur_diele_eg_battery
      - sensor.fenster_treppenhaus_battery_level
  - name: '{count} Heizungen: {range}%'
    icon: mdi:radiator
    min: 20
    entities:
      - binary_sensor.wandthermostat_buro_low_bat
      - binary_sensor.wandthermostat_wohnzimmer_low_bat
      - binary_sensor.heizung_buro_low_bat
      - sensor.heizung_bad_eg_battery_level
      - sensor.heizung_wc_battery_level
  - name: '{count} Luftfeuchtesensoren: {range}%'
    icon: mdi:water-percent
    min: 20
    entities:
      - sensor.luftfeuchtesensor_bad_eg_batterie
  - name: '{count} Rauchmelder: {range}%'
    icon: mdi:water-percent
    min: 20
    entities:
      - sensor.rauchmelder_buro_power
      - sensor.rauchmelder_eg_treppenhaus_power
      - sensor.rauchmelder_flur_power
      - sensor.rauchmelder_esszimmer_power
      - sensor.rauchmelder_hauswirtschaftsraum_power
      - sensor.rauchmelder_wohnzimmer_power
  - name: '{count} Schalter: {range}%'
    icon: mdi:water-percent
    min: 20
    entities:
      - sensor.wandschalter_wc_battery
      - sensor.wandschalter_hauswirtschaftsraum_batterie
      - sensor.remote_spule_kuche_battery
      - sensor.remote_arbeitsplatte_kuche_battery
      - sensor.remote_spule_kuche_battery
state_map:
  - from: High
    to: 100
  - from: Low
    to: 10
  - from: unavailable
    to: 0
  - from: 'on'
    to: 100
  - from: 'off'
    to: 10
tap_action:
  action: more-info

```

**Screenshots**
![image](https://github.com/maxwroc/battery-state-card/assets/54850893/55fed172-0ce8-48a0-8dcc-e9dda5b291d5)

**Version**
battery-state-card: 3.1.2
HA: 2024.1.2 (the issue seems to be independent of the HA version).

**Comment by maxwroc** [comment]
Oh true, It's because of the default state value formatting which was introduced in v3.1.0. 

You can mitigate it by disabling this feature via:
```yaml
default_state_formatting: false
```

The reason why you think it's not deterministic is that it affects only entities with decimal values :) Now card shows correct decimal point (according to locale settings). In your case it is comma but later cards code has problem converting this back to the JS number hance grouping min/max/range stops working.

Permanent fix should be easy although I would like to have a test covering this particular case (to avoid similar problems in the future).

Thanks for reporting the issue

**Comment by rockberte** [comment]
Aah - yeah that makes sence! Thx for the explanation.

I tried the mentioned setting and with that setting the issue is gone.

**Comment by maxwroc** [comment]
Fixed in v3.1.4
